nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -120,6 +120,14 @@ class BtServiceNode : public BT::SyncActionNode
   }
 
 protected:
+  void increment_recovery_count()
+  {
+    int recovery_count = 0;
+    config().blackboard->get<int>("number_recoveries", recovery_count);  // NOLINT
+    recovery_count += 1;
+    config().blackboard->set<int>("number_recoveries", recovery_count);  // NOLINT
+  }
+
```

Comment:
May be making some helper class or something similar would reduce the code.

Reply:
I think this _is_ the helper object (function) to do this for `N` nodes.

Reply:
I understand that you made for N nodes. But I saw the same function in other class too. So, I though of putting it in a helper class or something.

Reply:
But if this if you think it won't be used in future than I guess it would be fine.

Reply:
I’m just not sure where to stick it. There’s already a great deal of analog components between the service and action nodes this seems in theme. I think something like a bt utils file would be useful if we could identify more things we could pull out.

Reply:
Exactly, this is the best place I could find to make this part more usable. I think we can keep the function here till the point there's a bunch of stuff which can be moved to a common space.